### PR TITLE
Add conditional middleware example

### PR DESCRIPTION
**Pull request by boekkooi**
Read the code.

### CODE REVIEW

**Comment by rosstuck** [comment]
Legit pattern! Can you give a more specific use case for why you'd do this though? I feel in practice you wouldn't go bus->bus very often, but more bus->handler or bus->queue or bus->network.

**Comment by boekkooi** [comment]
@rosstuck One of the cases where I used this in the past was when there was one "main" CommandBus that was used for multiple type of commands. 

For example the default command is using db connection 1 the second command type is using db connection 2 and the final is using a external message queue. So for the DB connections we needed to add transaction and locking but for the message queue we just needed to log and transform the command and add it to a message queue.

This can be done using a handler but that would mean that you can't take advantage of default middleware so the bus->bus implementation is the best one for these cases. It's a minimal overhead for the maximum gain.

Hope this makes sence.

**Comment by rosstuck** [comment]
Hmm, would you be willing to rename the example to Command Routing instead of Conditional Middleware? I totally see that you can do both with this and think folks would get it, it might just zero in on the end goal more.

File: examples/6-conditional-handlers.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+use League\Tactician\StandardCommandBus;
+use League\Tactician\Middleware;
+
+// External command
+interface ExternalCommand extends Command
+{
+}
+
+// We create some custom middleware that will forward any ExternalCommand to
+// a separate CommandBus.
```
Comment (rosstuck) [inline examples/6-conditional-handlers.php]: Suggested text addition: "...or HTTP endpoint or queue." Make certain they realize it doesn't have to be another CommandBus

File: examples/6-conditional-handlers.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+final class ExternalCommandMiddleware implements Middleware
+{
+    /**
+     * @var CommandBus
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBus $commandBus
+     */
```
Comment (rosstuck) [inline examples/6-conditional-handlers.php]: You can probably drop the docblocks here for brevity. It's pretty clear from the naming and typehints.

File: examples/6-conditional-handlers.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+class MyExternalCommand implements ExternalCommand
+{
+}
+
+// Now  let's instantiate our ExternalCommandHandler, StandardCommandBus and ExternalCommandMiddleware
+$externalCommandHandler = new ExternalCommandHandler();
+$externalCommandBus = new StandardCommandBus([$externalCommandHandler]);
+$externalMiddleware = new ExternalCommandMiddleware($externalCommandBus);
+
+// Time to create out main CommandBus
```
Comment (rosstuck) [inline examples/6-conditional-handlers.php]: Our, not out. :smile:

File: examples/6-conditional-handlers.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+$commandBus = new \League\Tactician\StandardCommandBus([$externalMiddleware, $handlerMiddleware]);
```
Comment (rosstuck) [inline examples/6-conditional-handlers.php]: FQCN here but it's already imported via use at the top.

**Comment by boekkooi** [comment]
@rosstuck PR is updated

**Pull request merged**